bin/i18n/sync-out.rb

```diff
@@ -266,7 +266,9 @@ def sort_and_sanitize(hash)
   hash.sort_by {|key, _| key}.each_with_object({}) do |(key, value), result|
     case value
     when Hash
-      result[key] = sort_and_sanitize(value) unless value.empty?
+      # ensure we always call sort_and_sanitize on the hash to avoid top-level empty objects
+      sorted_hash = sort_and_sanitize(value)
```

Comment:
We need this to filter hash keys with `nil` values like `{ foo: nil }`, correct?